### ISSUE TITLE
Extra exercises not unlocking correctly

### ISSUE THREAD

**Issue by hilary** [exercism/exercism.io#3873]
I'm going to leave out the story of how I discovered this issue. Ask if you want to know...

TL;DR Extra exercises aren't unlocking correctly on multiple tracks (could be one issue, could be multiple.) I have completed about half the core exercises for Golang and have access to less than 1/3 of the extra exercises that should have unlocked.

* My data is for the go track
* #3869 (C#)
* #3867 (Delphi)

In more depth:

I compared the extra exercises I should have unlocked (on the basis of the go track [config.json](https://github.com/exercism/go/blob/master/config.json) to those I actually do have unlocked at the time I filed this issue. Here's what I learned:
* Of the nine core exercises I completed, six of them unlocked exactly one extra exercise each.
* All the extra exercises that have no prerequisite are unlocked.
* No extra exercise has been prematurely unlocked, i.e., all incorrect results are always locked when should be unlocked, never the opposite.
* Approval does not seem to matter. While I completed most of the core exercises without approval (of necessity!), I did receive approval for Hamming. One of the three extra exercises that Hamming unlocks is unlocked, the other two are locked. Note that the approval for Hamming predated exercism.v2 going live by a day or two, but showed up as an approval in the v2 interface.

**Comment by iHiD** [comment]
@hilary and I discussed this at length privately. 

The fundamental issue was a misunderstanding about unlocking for unapproved exercises. The misunderstanding was due to missing and incorrect copy on the site which I've fixed here: https://github.com/exercism/website/pull/183

> Approval does not seem to matter. While I completed most of the core exercises without approval (of necessity!), I did receive approval for Hamming. One of the three extra exercises that Hamming unlocks is unlocked, the other two are locked. Note that the approval for Hamming predated exercism.v2 going live by a day or two, but showed up as an approval in the v2 interface.

I need to look into this example as this seems broken.

**Comment by iHiD** [comment]
I'm (vaguely/pretty) confident that the unlocking behaviour is working now and that the data is all in a better shape, so I'm going to close this, as everything will have moved on since this was posted, and there are lots of similar issues with related problems. Please feel free to reopen if you think that's a bad call though :)

**Comment by hilary** [comment]
Sounds good to me 😺
